### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by ineiti**
@SuperQuentin @A2va:
- initial commit

@ineiti
- formatting files
- testing CMakeLists.txt
- updating README and some comments
- fixing warnings
- github workflows for formatting and warnings

### CODE REVIEW

File: src/input.tpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#include <iostream>
+#include <limits>
+#include <string>
+#include "../include/input.h"
```
Comment (A2va) [inline src/input.tpp]: No need to include `input.h` here. `input.tpp` is already include here https://github.com/HEIGVD-PRG1-F-2022/prg1f-io/blob/1d656f333abc34fc8ae55a4c952bf4a2bd6568ae/include/input.h#L26

Reference: https://stackoverflow.com/questions/495021/why-can-templates-only-be-implemented-in-the-header-file

File: include/display.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
+//
+// Project name : LAB 05 - Refactor
+// Project members :
+// - Florian Duruz
+// - Aellen Quentin
+//
+
+#ifndef REFACTOR_DISPLAY_H
+#define REFACTOR_DISPLAY_H
```
Comment (A2va) [inline include/display.h]: Change `REFACTOR_DISPLAY_H` to something like `DISPLAY_HEADER`

Comment (ineiti) [reply include/display.h]: Done

File: src/input.tpp (hunk 1 above)
Comment (ineiti) [reply src/input.tpp]: Done

**Pull request merged**